dhcpv4/nclient4/lease.go

```diff
@@ -0,0 +1,55 @@
+// This is lease support for nclient4
+
+package nclient4
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/insomniacslk/dhcp/dhcpv4"
+)
+
+// Lease contains a DHCPv4 lease after DORA.
+// note: Lease doesn't include binding interface name
+type Lease struct {
+	Offer        *dhcpv4.DHCPv4
+	ACK          *dhcpv4.DHCPv4
+	CreationTime time.Time
+}
+
+// NewReleaseFromLease creates a DHCPv4 Release message from the lease.
+// default Release message without any Modifer is created as following:
+//  - option Message Type is Release
+//  - ClientIP is set to lease.ACK.YourIPAddr
+//  - ClientHWAddr is set to lease.ACK.ClientHWAddr
+//  - Unicast
+//  - option Server Identifier is set to ServerIdentifier of lease.ACK
+func NewReleaseFromLease(lease *Lease, modifiers ...dhcpv4.Modifier) (*dhcpv4.DHCPv4, error) {
```

Comment:
I was suggesting to add a `NewReleaseFromACK` in the main dhcp lib similar to `NewRequestFromOffer`.
Anyway, this lgtm

Reply:
I knew, and then I realized that if I put `NewReleaseFromACK` in dhcpv4 pkg, then dhcpv4 need to import nclient4 pkg since Lease is defined in nclient4, then it is import loop since nclient4 also import dhcpv4; to solve that, I need to move Lease to dhcpv4, then I feel it is a bit odd for dhcpv4 to include Lease since dhcpv4 is supposed to be protocol level only pkg while Lease is really a client thing

Reply:
You do not need to import the `lease` struct to have a base `NewReleaseFromACK` or `NewReleaseFromRequest`.

```
func NewReleaseFromACK(ack *DHCPv4, modifiers ...Modifier) (*DHCPv4, error) {
```

And then you can use that function here.

Reply:
ah, sorry, I just realized you were talking abut ACK not Lease, I don't know what happened, but somehow my mind automatically translated NewReleaseFromACK to NewReleaseFromLease ... 
I could actually change it to NewReleaseFromACK and put it in dhcpv4